### PR TITLE
GLT-490 Added tests for SQLStudyDAO

### PR DESCRIPTION
**Pull request by tonydebat**
Updated hamcrest dependency.

### CODE REVIEW

File: sqlstore/src/test/java/uk/ac/bbsrc/tgac/miso/sqlstore/SQLStudyDAOTest.java [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+  @Test
+  public void testListBySearchNone() {
+    List<Study> runs = dao.listBySearch("pizza");
+    assertEquals(0, runs.size());
+  }
+
+  @Test
+  public void testListBySearchEmpty() {
+    List<Study> studies = dao.listBySearch("");
```
Comment (apmasell) [inline sqlstore/src/test/java/uk/ac/bbsrc/tgac/miso/sqlstore/SQLStudyDAOTest.java]: This seems to liberal as a test.

Comment (apmasell) [inline sqlstore/src/test/java/uk/ac/bbsrc/tgac/miso/sqlstore/SQLStudyDAOTest.java]: I guess: if someone searches for empty string, should they get results at all?

**Comment by apmasell** [comment]
:+1:

**Comment by frankie13** [comment]
@tonydebat I don't think your test is getting run as part of the suite.

There is no change to the **TestSuite file and this test file doesn't end with Suite so it won't get run.

File: sqlstore/src/test/java/uk/ac/bbsrc/tgac/miso/sqlstore/SQLStudyDAOTest.java (hunk 1 above)
Comment (tonydebat) [inline sqlstore/src/test/java/uk/ac/bbsrc/tgac/miso/sqlstore/SQLStudyDAOTest.java]: The search adds a SQL wildcard %. So searching for something that doesn't exist returns nothing and searching for nothing returns everything.

**Pull request merged**